tiledb/sm/filesystem/mem_filesystem.cc

```diff
@@ -213,37 +338,22 @@ class MemFilesystem::Directory : public MemFilesystem::FSNode {
 MemFilesystem::MemFilesystem()
     : root_(std::unique_ptr<Directory>(new Directory("", nullptr))) {
   assert(root_);
-};
+}
 
-MemFilesystem::~MemFilesystem(){};
+MemFilesystem::~MemFilesystem() {
+}
 
 Status MemFilesystem::create_dir(const std::string& path) const {
-  std::unique_lock<std::mutex> fs_lock(memfilesystem_mutex_);
-  return create_dir_unsafe(path);
+  return create_dir_internal(path);
```

Comment:
do we need those internal methods (create_dir_internal , touch_internal) now that they are the same as the externals?

Reply:
Good question. I thought about this too, but it would require the new output argument (FSNode**) to be in the public API, which would force us to make FSNode a public class instead of private (as it is now).